homeassistant/components/sensor/onewire.py

```diff
@@ -0,0 +1,93 @@
+""" Support for DS18B20 One Wire Sensors"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELCIUS, TEMP_FAHRENHEIT
+from glob import glob
+import os
+import time
+import logging
+
+
+BASE_DIR = '/sys/bus/w1/devices/'
+DEVICE_FOLDERS = glob(os.path.join(BASE_DIR, '28*'))
+SENSOR_IDS = [os.path.split(device_folder)[1] for device_folder in DEVICE_FOLDERS]
+DEVICE_FILES = [os.path.join(device_folder, 'w1_slave') for device_folder in DEVICE_FOLDERS]
+
+_LOGGER = logging.getLogger(__name__)
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the one wire Sensors"""
+    # TODO check if kernel modules are loaded
+
+    # TODO implment config fore the name, but also default solution
+    if DEVICE_FILES == []:
+        _LOGGER.error('No onewire sensor found')
+        return
+
+    devs = []
+    names = []
+    try:
+        ## only one name given
+        if type(config['names']) == str:
+            names = [names]
+        
+        ## map names and sensors in given order
+        elif type(config['names']) == list:
+            names = config['names']
+
+        ## map names with ids
+        elif type(config['names']) == dict:
+            for sensor_id in SENSOR_IDS:
```

Comment:
You could fallback to use the id as the name if the id key is not found in the dict:
```python
names = [config['names'].get(sensor_id, sensor_id) for sensor_id in SENSOR_IDS]
```
(the second parameter to get is the fallback value if key not found)